streams/runners/storm/runtime/src/main/java/com/hortonworks/streamline/streams/runtime/storm/bolt/kafka/StreamlineEventSerializer.java

```diff
@@ -67,87 +78,41 @@ public void close() {
         }
     }
 
-    private SchemaMetadata createSchemaMetadata(String topic) {
-        SchemaMetadata schemaMetadata = getSchemaKey(topic, false);
-        String description = "Schema registered by KafkaAvroSerializer for topic: [" + topic + "] iskey: [" + false + "]";
-        return new SchemaMetadata.Builder(schemaMetadata).description(description).compatibility(compatibility).build();
-    }
-
-    protected SchemaMetadata getSchemaKey(String topic, boolean isKey) {
-        return Utils.getSchemaKey(topic, isKey);
-    }
-
-    private static GenericRecord getAvroGenericRecord (Map<String, Object> streamlineEvent, String topic) {
-        GenericRecord result = null;
-        if (streamlineEvent != null && !streamlineEvent.isEmpty()) {
-            result = (GenericRecord) getAvroSchemaAndValue(streamlineEvent, SCHEMA_NAMESPACE, topic).get(VALUE_KEY);
+    //package level access for testing
+    static Object getAvroRecord (StreamlineEvent streamlineEvent, Schema schema) {
+        if (streamlineEvent.keySet().size() == 1 && streamlineEvent.containsKey(StreamlineEvent.PRIMITIVE_PAYLOAD_FIELD)) {
```

Comment:
It may be better to have check like below to handle invalid schema scenario also with .

```
if (streamlineEvent.containsKey(StreamlineEvent.PRIMITIVE_PAYLOAD_FIELD)) {
  if((streamlineEvent.keySet().size() > 1 ) {
    throw new RuntimeException("Invalid schema, primitive schema can contain only one field."); // can have better error message if any
  }
  return streamlineEvent.get(StreamlineEvent.PRIMITIVE_PAYLOAD_FIELD);
}
```